integration/benchmark/main.go

```diff
@@ -0,0 +1,338 @@
+
+	loaderMetrics.SetTPSConfigured(loadCase.tps)
+
+	var lg *utils.ContLoadGenerator
+	lg, err = utils.NewContLoadGenerator(
+		log,
+		loaderMetrics,
+		flowClient,
+		supervisorClient,
+		loadedAccessAddr,
+		priv,
+		&serviceAccountAddress,
+		&fungibleTokenAddress,
+		&flowTokenAddress,
+		loadCase.tps,
+		accountMultiplier,
+		utils.LoadType(loadType),
+		feedbackEnabled,
+		utils.ConstExecParam{
+			MaxTxSizeInByte: *maxConstExecTxSizeInBytes,
+			AuthAccountNum:  *authAccNumInConstExecTx,
+			ArgSizeInByte:   *argSizeInByteInConstExecTx,
+			PayerKeyCount:   *payerKeyCountInConstExecTx,
+		},
+	)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("unable to create new cont load generator")
+	}
+
+	err = lg.Init()
+	if err != nil {
+		log.Fatal().Err(err).Msgf("unable to init loader")
+	}
+	lg.Start()
+
+	testStartTime := time.Now()
+	time.Sleep(loadCase.duration)
+	testEndTime := time.Now()
+
+	if lg != nil {
+		lg.Stop()
+	}
+
+	dataSlices := calculateTpsSlices(
+		testStartTime,
+		testEndTime,
+		*leadTime,
+		*sliceSize,
+		gitSha,
+		goVersion,
+		osVersion,
+		lg)
+	prepareDataForBigQuery(dataSlices, *ciFlag)
+}
+
+func calculateTpsSlices(start, end time.Time, leadTime, sliceTime, commit, goVersion, osVersion string, lg *utils.ContLoadGenerator) []dataSlice {
+	//remove the lead time on both start and end, this should remove spin-up and spin-down times
+	leadDuration, _ := time.ParseDuration(leadTime)
+	endTime := end.Add(-1 * leadDuration)
+	sliceDuration, _ := time.ParseDuration(sliceTime)
+
+	slices := make([]dataSlice, 0)
+
+	for currentTime := start.Add(leadDuration); currentTime.Add(sliceDuration).Before(endTime); currentTime = currentTime.Add(sliceDuration) {
+		sliceEndTime := currentTime.Add(sliceDuration)
+
+		inputTps := lg.AvgTpsBetween(currentTime, sliceEndTime)
+		proStart := getPrometheusTransactionAtTime(currentTime)
+		proEnd := getPrometheusTransactionAtTime(sliceEndTime)
+
+		outputTps := (proEnd - proStart) / (sliceDuration).Seconds()
+
+		slice := dataSlice{
+			GitSha:              commit,
+			GoVersion:           goVersion,
+			OsVersion:           osVersion,
+			StartTime:           currentTime,
+			EndTime:             currentTime.Add(sliceDuration),
+			InputTps:            inputTps,
+			OutputTps:           outputTps,
+			ProStartTransaction: proStart,
+			ProEndTransaction:   proEnd}
+
+		slices = append(slices, slice)
+	}
+
+	return slices
+}
+
+func prepareDataForBigQuery(slices []dataSlice, ci bool) {
+	jsonText, err := json.Marshal(slices)
+	if err != nil {
+		println("Error converting slice data to json")
+	}
+
+	// if we are uploading to BigQuery then we need a copy of the json file
+	// that is newline delimited json
+	if ci {
+		jsonString := string(jsonText)
+		// remove the surrounding square brackets
+		jsonString = jsonString[1 : len(jsonString)-1]
+		// end of object commas will be replaced with newlines
+		jsonString = strings.ReplaceAll(jsonString, "},", "}\n")
```

Comment:
you should likely just do:
```
for _, slice := range slices {
  jsonText, _ := json.Marshal(slice)
  ...
}
```

This will avoid the text mangling.

Reply:
This would still give us square brackets that would need to be removed on each object and we would have to remarshal for the non-newline JSON file anyway. It ends up replacing just line 269 with a loop. Are we sure we want this change?